core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -320,14 +321,15 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         .filter { entry =>
           try {
             val prevFileSize = fileToAppInfo.get(entry.getPath()).map{_.fileSize}.getOrElse(0L)
+            fs.access(entry.getPath, FsAction.READ)
```

Comment:
This API is tagged with `  @LimitedPrivate({"HDFS", "Hive"})`, but I think it should be fine to call it here.